WordPress/Classes/ViewRelated/Reader/Controllers/ReaderStreamViewController+Helper.swift

```diff
@@ -107,42 +104,105 @@ extension ReaderStreamViewController {
-                                                                           comment: "Message displayed in Reader Saved Posts view if a user hasn't yet saved any posts."),
-                                                  message: NSLocalizedString("Tap [bookmark-outline] to save a post to your list.",
-                                                                             comment: "A hint displayed in the Saved Posts section of the Reader. The '[bookmark-outline]' placeholder will be replaced by an icon at runtime – please leave that string intact."))
+    @ViewBuilder
+    private func _makeEmptyStateView(_ type: EmptyStateViewType) -> some View {
+        switch type {
+        case .steamLoadingFailed:
+            EmptyStateView(
+                ResultsStatusText.loadingErrorTitle,
+                systemImage: "exclamationmark.circle",
+                description: ResultsStatusText.loadingErrorMessage
+            )
+        case .noSavedPosts:
+            EmptyStateView(label: {
+                Label(NSLocalizedString("No saved posts", comment: "Message displayed in Reader Saved Posts view if a user hasn't yet saved any posts."), image: "wp-illustration-reader-empty")
+            }, description: {
+                // Had to use UIKit because Text(AttributedString()) won't render the attachment
+                HostedAttributedLabel(text: self.makeSavedPostsEmptyViewDescription())
+                    .fixedSize()
+            }, actions: {
+                EmptyView()
+            })
+        case .discover:
+            EmptyStateView(
+                ReaderStreamViewController.defaultResponse.title,
+                image: "wp-illustration-reader-empty",
+                description: ReaderStreamViewController.defaultResponse.message
+            )
+        case .noConnection:
+            EmptyStateView(
+                ResultsStatusText.noConnectionTitle,
+                systemImage: "network.slash",
+                description: noConnectionMessage()
+            )
+        case .noFollowedSites:
+            EmptyStateView(label: {
+                Label(Strings.noFollowedSitesTitle, systemImage: "checkmark.circle")
+            }, description: {
+                Text(Strings.noFollowedSitesSubtitle)
+            }, actions: {
+                Button(Strings.noFollowedSitesButtonTitle) {
+                    RootViewCoordinator.sharedPresenter.showReader(path: .discover)
+                }
+                .buttonStyle(.primary)
+            })
+        }
+    }
 
-        var messageText = NSMutableAttributedString(string: noResultsResponse.message)
+    private func makeSavedPostsEmptyViewDescription() -> NSAttributedString {
+        let details = NSLocalizedString("Tap [bookmark-outline] to save a post to your list.", comment: "A hint displayed in the Saved Posts section of the Reader. The '[bookmark-outline]' placeholder will be replaced by an icon at runtime – please leave that string intact.")
```

Comment:
This special string puts a lot of trust into the translation process. I'm surprised there is only one translation that didn't follow the instruction of keeping `[bookmark-outline]`. I'm not familiar with the translation process. How can it be fixed?

```
// WordPress/Resources/ja.lproj/Localizable.strings
6112:"Tap [bookmark-outline] to save a post to your list." = "ブックマークアイコンをタップし、投稿をブックマークリストに保存します。";
```

Reply:
I was initially going to remove it. If it's only one localization, I'd ignore it for now. It should've probably used `%@` instead of `[bookmark-outline]` as translators know how to work with these placeholders.